velox/dwio/parquet/reader/ParquetReader.cpp

```diff
@@ -629,7 +635,17 @@ ParquetRowReader::ParquetRowReader(
   if (rowGroups_.empty()) {
     return; // TODO
   }
-  ParquetParams params(pool_, columnReaderStats_, readerBase_->fileMetaData());
+
+  if (readerBase_->isUseParquetBloomFilter()) {
+    bloomFilterReader_ = BloomFilterReader::make(
```

Comment:
If there is no bloom filter in any row group or column chunk, there is no need to create a bloomFilterReader_. 
Actually, I wonder why Arrow used so many classes BloomFilterReader, BloomFilterReaderImpl, RowGroupBloomFilterReader, RowGroupBloomFilterReaderImpl and BloomFilter. It adds several levels to trace when people reads the source code and not quite strait-forward. The only important and necessary structure is BloomFilter. Can we just have a simple function in ParquetData
```
std::unique_ptr<BloomFilter> getBloomFilter(const thrift::RowGroup& rowgroup, uint32_t column)
```
that just read the bloom filter for the specified rowgroup and column? Then we can remove bloomFilterReader_ and only lazily read the BloomFilter in `ParquetData::rowGroupMatches(uint32_t rowGroupId, common::Filter* FOLLY_NULLABLE filter)` and no need to pass in the rowGroupBloomFilterReader

Reply:
We need file input stream to create the bloom filter reader which is not available in ParquetData. Will create the BloomFilterReader, call the function to get the RowGroupBloomFilterReader only when required.